### DISCUSSION TITLE
mecodev gamma and beta diversity calculations

### DISCUSSION THREAD

**Discussion by jamorillo** [ChiLiubio/microeco#420] [General]
Hi Chi,

I have a question regarding the mecodev package. I noticed that in one of the steps it calculates both beta and gamma diversity using a microeco object, which is very useful. However, I would like to clarify a few points about these numbers:

1) I understand that the gamma diversity calculation refers to an estimation of the total number of ASVs detected in a group of samples (Site). Is that correct? Is there any kind of estimation involved, or are these the total detected ASVs/group?

2) Regarding beta diversity, do these numbers represent the average distance (Bray in this case) within each group? Is it something similar/equivalent to betadisper?

3) Can I apply apply sqrt to calculate betadiversity? then giving more weight to less frequent features. Might be interesting. How?

3) Which paper should I cite if I use mecodev?

Many thanks!
reagrds
Jose

Example with one of my datasets:

> microecoITS_norm
microtable-class object:
sample_table have 156 rows and 18 columns
otu_table have 3128 rows and 156 columns
tax_table have 3128 rows and 8 columns
phylo_tree have 3128 tips
rep_fasta have 3128 sequences

> diversity_object <- trans_gamma$new(dataset = microecoITS_norm, 
                                         group = "Site", 
                                         method = "bray")

> diversity_object$cal_observed()
The result is stored in object$res_observed ...

> diversity_object$res_observed
    Sample Group      beta gamma
1 observed    FR 0.6718374  1637
2 observed    DP 0.4799911   172
3 observed    MC 0.6504501   705
4 observed    BG 0.6054494   763
5 observed    NR 0.6180772   514

**Comment by ChiLiubio** [comment]
Hi Jose,
1. The gamma diversity here is actually the number of total species observed, no estimation.
2. Yes. It is the mean dispersion from the centroid of distance matrix based on the `betadisper` as you say. Thanks. I will explain these in the package documentation.
3. Sure. You can. 
4. No need to cite the paper. Just provide a link to the package.

Best,
Chi

**Comment by jamorillo** [comment]
Hi Chi, 
Thanks for the confirmation of betadisper. One thing more, respect to tansformations of ASV table (sqrt, log etc), is that integrated in microeco, perhaps before (I guess) than cal_betadiv? 
thanks!
PD: I think that found a way:


microecoITS_norm$cal_abund()

### transformation  here:

microecoITS_norm$taxa_abund$ASVs <- sqrt(microecoITS_norm$taxa_abund$ASVs)

microecoITS_norm$cal_betadiv(unifrac = FALSE)

beta_ITS <- trans_beta$new(dataset = microecoITS_norm, 
                           group = "Site",
                           measure = "bray")

**Reply by ChiLiubio** [reply]
Hi. It should be :
```
microecoITS_norm$otu_table <- sqrt(microecoITS_norm$otu_table)
```
